src/Nethermind/Nethermind.Trie/PatriciaTree.cs

```diff
@@ -710,15 +730,18 @@ private void ConnectNodes(TrieNode? node, in TraverseContext traverseContext)
                                 }
                             }
 
-                            TrieNode childNode = node.GetChild(TrieStore, childNodeIndex);
+                            node.AppendChildPathBranch(ref path, childNodeIndex);
+                            TrieNode childNode = node.GetChildWithChildPath(TrieStore, ref path, childNodeIndex);
                             if (childNode is null)
                             {
                                 /* potential corrupted trie data state when we find a branch that has only one child */
                                 throw new TrieException(
                                     "Before updating branch should have had at least two non-empty children");
                             }
 
-                            ResolveNode(childNode, in traverseContext);
+                            ResolveNode(childNode, in traverseContext, in path);
+                            path.TruncateOne();
```

Comment:
I'm not sure I like the mutability here, that requires to provide pop behavior whenever we append the path. As mentioned in another comment, it could be either this with value semantics, or copy on append (Paprika took this was on `NibblePath.Append`) to an externally provided memory or maybe some `using` to declare scope? On the other hand if this works...

Reply:
Tried both. Originally with just copying and then a `using` wrapper, see `TreePath.AppendScope`. Tragically, they are about 5-10% overhead.

Reply:
Oh, probably the finally overhead 😩 Then nothing to do here.